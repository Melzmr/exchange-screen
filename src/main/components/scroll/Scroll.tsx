import React, {CSSProperties} from 'react';

export interface IScrollProps {
  className?: string;
  style?: CSSProperties;
}

export const Scroll: React.FC<IScrollProps> = ({children, className, style}) => {
  return (
      <div
          className={className}
          style={style}
      >
        {children}
      </div>
  );
};

Scroll.displayName = 'Scroll';
