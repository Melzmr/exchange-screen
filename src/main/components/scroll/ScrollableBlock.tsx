import React from 'react';
import Slider, {Settings} from 'react-slick';

export interface IScrollableBlockProps extends Settings {
  className?: string;
}


export const ScrollableBlock: React.FC<IScrollableBlockProps> = ({children, ...props}) => {

  const settings: Settings = {
    dots: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    className: 'responsive_block',
    accessibility: true,
    arrows: false,
  };

  return (
      <Slider
          {...settings}
          {...props}
          // className="responsive_block"
      >
        {children}
      </Slider>
  );
};

ScrollableBlock.displayName = 'ScrollableBlock';
