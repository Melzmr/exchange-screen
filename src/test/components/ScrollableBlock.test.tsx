import React from 'react';
import {shallow} from 'enzyme';
import {ScrollableBlock} from '../../main/components/scroll/ScrollableBlock';

describe('ScrollableBlock', () => {
  let comp: React.ReactElement;

  beforeEach(() => {
    comp = <ScrollableBlock/>;
  });

  it('renders ScrollableBlock component', () => {
    const wrapper = shallow(comp);
    expect(wrapper).toMatchSnapshot();
  });

});


