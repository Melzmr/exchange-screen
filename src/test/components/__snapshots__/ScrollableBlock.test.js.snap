// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableBlock renders ScrollableBlock component 1`] = `
<Slider
  accessibility={true}
  arrows={false}
  className=""
  dots={true}
  infinite={true}
  slidesToScroll={1}
  slidesToShow={1}
  speed={500}
/>
`;
